python/lsst/pipe/tasks/imageDifference.py

```diff
@@ -1277,6 +1277,75 @@ def _makeArgumentParser(cls):
         return parser
 
 
+class ImageDifferenceFromTemplateConnections(ImageDifferenceTaskConnections,
+                                             defaultTemplates={"coaddName": "goodSeeing"}
+                                             ):
+    inputTemplate = pipeBase.connectionTypes.Input(
+        doc=("Warped template produced by GetMultiTractCoaddTemplate. ",
+             "Replaces need for coaddExposures"),
+        dimensions=("instrument", "visit", "detector"),
+        storageClass="ExposureF",
+        name="{fakesType}{coaddName}Diff_templateExp{warpTypeSuffix}",
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+        # ImageDifferenceConnections will have removed one of these.
+        # Make sure they're both gone. No coadds are needed.
+        if "coaddExposures" in self.inputs:
+            self.inputs.remove("coaddExposures")
+        if "dcrCoadds" in self.inputs:
+            self.inputs.remove("dcrCoadds")
+
+
+class ImageDifferenceFromTemplateConfig(ImageDifferenceConfig,
+                                        pipelineConnections=ImageDifferenceFromTemplateConnections):
+    pass
+
+
+class ImageDifferenceFromTemplateTask(ImageDifferenceTask):
+    ConfigClass = ImageDifferenceFromTemplateConfig
+    _DefaultName = "imageDifference"
+
+    @lsst.utils.inheritDoc(pipeBase.PipelineTask)
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        self.log.info("Processing %s", butlerQC.quantum.dataId)
+        self.checkTemplateIsSufficient(inputs['inputTemplate'])
+        expId, expBits = butlerQC.quantum.dataId.pack("visit_detector",
+                                                      returnMaxBits=True)
+        idFactory = self.makeIdFactory(expId=expId, expBits=expBits)
+
+        outputs = self.run(exposure=inputs['exposure'],
+                           templateExposure=inputs['inputTemplate'],
+                           idFactory=idFactory)
+
+        # Consistency with runDataref gen2 handling
+        if outputs.diaSources is None:
+            del outputs.diaSources
+        butlerQC.put(outputs, outputRefs)
+
+    def checkTemplateIsSufficient(self, templateExposure):
+        """Raise NoWorkFound if template coverage < requiredTemplateFraction
+
+        templateExposure : `lsst.afw.image.ExposureF`
```

Comment:
Please finish formatting the docstring (labeled `Parameters` and `Raises` sections)